supplier_product/models.py

```diff
@@ -0,0 +1,35 @@
+from django.db import models
+from django.core.validators import MinValueValidator
+from product.models import Product
+from supplier.models import Supplier
+
+
+class SupplierProduct(models.Model):
+    suppler_product_id = models.IntegerField(primary_key=True)
+    qr_code = models.ForeignKey(Product, on_delete=models.CASCADE)
+    user_name = models.ForeignKey(Supplier, on_delete=models.CASCADE)
+    price = models.IntegerField(validators=[MinValueValidator(1)])
+    quantity = models.IntegerField()
+
+    class Meta:
+        unique_together = ('user_name', 'qr_code')
+
+    @staticmethod
+    def save_sup_product(sup_prod):
+        SupplierProduct.full_clean(sup_prod)
+        sup_prod.save()
+
+    @staticmethod
+    def delete_sup_product(self):
+        self.delete()
+
+    def increase_sup_product_quantity(self, quantity):
+        self.quantity += quantity
+        self.save()
+
+    def decrease_sup_product_quantity(self, quantity):
```

Comment:
I don't understand this method. 
Why should we delete the record if self.quantity <= quantity?
Maybe we could return an error message, instead of deleting it?

Reply:
if we want to reduce the quantity to less then we have it means to delete the product , i believe we just delete it without error message

Reply:
See my comment [here](https://github.com/redhat-beyond/BuyTogether/pull/38#discussion_r1058219817).
I don't think we need to delete it if its quantity equals zero (or less).
I think we should return an error if the quantity we got is less than the current quantity.